tests/tests_pytorch/accelerators/test_tpu.py

```diff
@@ -323,3 +323,63 @@ def test_trainer_config_device_ids(devices, expected_device_ids):
     trainer = Trainer(accelerator="tpu", devices=devices)
     assert trainer.device_ids == expected_device_ids
     assert trainer.num_devices == len(expected_device_ids)
+
+
+@RunIf(tpu=True, standalone=True)
+@mock.patch.dict(os.environ, os.environ.copy(), clear=True)
+def test_all_gather_sync_grads(tmpdir):
```

Comment:
I recommend combining the two tests into one. Spawning processes is expensive, and here I see too little justification for it to be two separate ones. 

We will also need unit tests for the xla strategy in lightning_lite. I'm happy to help adding these if desired. 
(I suggest we use the xla_launch utility from tests/tests_lite/strategies/test_xla.py)

Reply:
Alright, I'll combine them and have a look at the test for lightning_lite. I'll ping you if I need a hand with it.